### PR TITLE
Package some DNS security tools

### PR DESCRIPTION
**Pull request by c0bw3b**
###### Motivation for this change
Refresh of #30542
And I'd like to have these tools available from nixpkgs

cc @globin @treemo 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by c0bw3b** [comment]
Result of `nix-review pr 61714` [1](https://github.com/Mic92/nix-review)
<details>
  <summary>7 package were build:</summary>

  - dnsenum
  - dnsrecon
  - fierce
  - perl528Packages.NetNetmask
  - perl528Packages.NetWhoisIP
  - perl528Packages.StringRandom
  - theharvester
</details>

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build perl528Packages.NetNetmask perl528Packages.NetWhoisIP perl528Packages.StringRandom dnsenum dnsrecon fierce theharvester

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build perl528Packages.NetNetmask perl528Packages.NetWhoisIP perl528Packages.StringRandom dnsenum dnsrecon fierce theharvester

**Comment by globin** [comment]
@c0bw3b btw github allows multiple authors on commits: https://help.github.com/en/articles/creating-a-commit-with-multiple-authors

File: pkgs/tools/security/dnsenum/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  installPhase = ''
+    install -vD dnsenum.pl $out/bin/dnsenum
+    install -vD dns.txt -t $out/share
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/fwaeytens/dnsenum;
+    description = "A tool to enumerate DNS information";
+    maintainers = [ maintainers.globin ];
```
[the remark this answers is not in the document]
Comment (c0bw3b) [reply pkgs/tools/security/dnsenum/default.nix]: Original work from his PR. I'll add myself too and rewrite those commits so @globin appears as co-author as suggested above :)

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -11426,6 +11426,18 @@ let
     };
   };
 
+  NetNetmask = buildPerlPackage rec {
+    name = "Net-Netmask-1.9104";
```
[the remark this answers is not in the document]
Comment (c0bw3b) [reply pkgs/top-level/perl-packages.nix]: I can change it but the rest of this file keeps things compact (less SLOC).
other perl packages are defined like this and are properly referenced by Repology like NetIP just above:
https://repology.org/project/perl:net-ip/versions

**Comment by c0bw3b** [comment]
> @c0bw3b btw github allows multiple authors on commits: https://help.github.com/en/articles/creating-a-commit-with-multiple-authors

Thanks, I did not know about this feature.
I'll rewrite those commits later today.

**Comment by c0bw3b** [comment]
Result of `nix-review pr 61714` [1](https://github.com/Mic92/nix-review)
<details>
  <summary>7 package were build:</summary>

  - dnsenum
  - dnsrecon
  - fierce
  - perl528Packages.NetNetmask
  - perl528Packages.NetWhoisIP
  - perl528Packages.StringRandom
  - theharvester
</details>

---
I've been using all 4 tools successfully too

**Comment by teto** [comment]
@c0bw3b ready for merge then ?

**Comment by c0bw3b** [comment]
Yes it is. :) Thanks @teto

**Comment by teto** [comment]
trusting you on this :p

**Pull request merged**